portality/models/account.py

```diff
@@ -152,12 +149,19 @@ def prep(self):
     @property
     def api_key(self):
         if self.has_role('api'):
-            return self.data.get('api_key', None)
+            # Return the stored api key if it's there, generate one if not (there should be one present if the role is)
```

Comment:
> as a side note, why can't I put a function call in the dict.get() method? when I tried to write self.data.get('api_key', self.generate_api_key()) it always ran the generate, whereas if the default arg was None it would return the key from the dict.

Hmm. Sorry, no idea beyond putting it in a temporary var.